dot/state/block_notify.go

```diff
@@ -26,22 +26,18 @@ import (
 	"github.com/google/uuid"
 )
 
-// RegisterImportedChannel registers a channel for block notification upon block import.
-// It returns the channel ID (used for unregistering the channel)
-func (bs *BlockState) RegisterImportedChannel(ch chan<- *types.Block) (byte, error) {
-	bs.importedLock.RLock()
+// DEFAULT_BUFFER_SIZE buffer size for channels
+const DEFAULT_BUFFER_SIZE = 100
 
-	id, err := bs.importedBytePool.Get()
-	if err != nil {
-		return 0, err
-	}
+// GetImportedBlockNotifierChannel function to retrieve a imported block notifier channel
+func (bs *BlockState) GetImportedBlockNotifierChannel() (chan *types.Block, error) {
```

Comment:
```suggestion
func (bs *BlockState) GetImportedBlockNotifierChannel() (<-chan *types.Block, error) {
```

Reply:
also, these functions don't identify the channel by an ID anymore - is the pointer to the channel the same in both functons? (I'm not sure if the read-only and write-only parts of the channel point to different locations potentially)